### PR TITLE
feat: Reduce bundle size by making caching optional

### PR DESCRIPTION
**Pull request by aubcel**
* replace `deep-map-key` with `camelcase-keys`
* remove `npmrc`
* make `axios-cache-adapter` an optional dependency
* remove `dist` from being commited

### CODE REVIEW

File: tests/unit/unit.test.ts [hunk 1]
```diff
@@ -45,7 +46,9 @@ describe("Clockodo (instance)", () => {
         it("throws an error when constructor has cacheTime with type other than number", () => {
```
Comment (aubcel) [inline tests/unit/unit.test.ts:46]: @jhnns I am confused why that test still passes on my machine. Even typescript doesn't complain. If I add the `cacheTime` to the test above typescript is complaning...
Do you know, how something like that happens?

File: src/plugins/cache.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import {setup} from "axios-cache-adapter";
```
Comment (aubcel) [inline src/plugins/cache.ts:1]: This file doesn't end up in the build `dist` folder...
Can you explain me why @jhnns ?

Comment (jannikkeye) [reply src/plugins/cache.ts:1]: Its not used anywhere except int the unit.test.js file which is not being built and put to dist.

Comment (aubcel) [reply src/plugins/cache.ts:1]: ah yeah, makes sense... yes i was reffering to that :)
Thanks

File: tests/unit/unit.test.ts (hunk 1 above)
Comment (aubcel) [reply tests/unit/unit.test.ts:46]: Bildschirmfoto 2020-07-08 um 10 35 33
But, yes removing the test makes sense anyway. Was just curious.

**Pull request merged**